terraform/root/aws-dm-dev/main.tf

```diff
@@ -8,3 +8,9 @@ module "aws_env" {
   aws_main_account_id = "${var.aws_main_account_id}"
   aws_dev_account_id = "${var.aws_dev_account_id}"
 }
+
+module "s3-bucket" {
```

Comment:
In this case the module name should be "dev_uploads_s3_bucket" or similar. Also it's a good rule to use underscores instead of dashes in module/resource names.